doc/build/snippets/index.rst

```diff
@@ -0,0 +1,25 @@
+.. _snippets:
+
+Snippets
+########
+
+Snippets are a way to save build system settings in one place, and then use
+those settings when you build any Zephyr application. This lets you save common
+configuration separately when it applies to multiple different applications.
+
```

Comment:
In my personal version of snippets, I have had good value from being able to add a `.cmake` file to be executed as well. This lets you remove separate secure and nonsecure boards.

nonsecure.cmake
```
list(APPEND DTS_EXTRA_CPPFLAGS "-DARM_TRUSTZONE_NON_SECURE")
```
nonsecure.conf
```
CONFIG_TRUSTED_EXECUTION_NONSECURE=y
```
nonsecure.overlay
```
/ {
	chosen {
		zephyr,code-partition = &nonsecure_partition;
		zephyr,sram = &sram0_ns;
	};
};
```
nrf9160dk_nrf9160.dts
```
#ifdef ARM_TRUSTZONE_NON_SECURE
#include <nordic/nrf9160ns_sica.dtsi>
#else
#include <nordic/nrf9160_sica.dtsi>
#endif /* ARM_TRUSTZONE_NON_SECURE */
```

`west build -b nrf9160dk_nrf9160 -Snonsecure zephyr/samples/hello_world`